charts/nextcloud/templates/deployment.yaml

```
@@ -350,16 +350,25 @@ spec:
         - name: nextcloud-config
           configMap:
             name: {{ template "nextcloud.fullname" . }}-config
+            {{- if .Values.nextcloud.configs.defaultMode }}
```

Please use `with`

```suggestion
            {{- with .Values.nextcloud.configs.defaultMode }}

            defaultMode: {{ . }}
            {{- end }}
```

PS: your current code would not work